src/lib/bundle.ts

```diff
@@ -0,0 +1,64 @@
+import glob from 'glob'
+import path from 'path'
+import { promisify } from 'util'
+import { warn } from './log'
+
+const aGlob = promisify(glob)
+
+interface Bundle {
+  name: string
+  scripts: string[]
+  pages: string[]
+}
+
+const sanitizeBundle = (bundle: Bundle): Bundle => {
+  interface Filename {
+    original: string
+    name: string
+    ext: string
+  }
+
+  const scriptsByName: { [name: string]: Filename | undefined } = {}
+  bundle.scripts.forEach(filename => {
+    const { ext, name } = path.parse(filename)
+
+    // check that file name (without ext) is unique (case-insensitive; don't allow Index.js and index.ts)
+    // and
+    // priorize extensions (ts > js)
+    const cannonicalName = name.toLowerCase()
+
+    const existingScript = scriptsByName[cannonicalName]
+    if (existingScript) { // a script with same canonical name was already seen
+      if (existingScript.ext !== ext) { // they have different extensions
+        warn(`Found a js file with the same name as ts file in bundle '${bundle.name}'. js file will be ignored (${existingScript.original} / ${filename}).`)
+        const shouldPreserveExistingTs = existingScript.ext === 'ts'
+        if (shouldPreserveExistingTs) { return }
+      } else { // same extension and filename ! probably a case issue
+        warn(`Found duplicate filename in bundle '${bundle.name}', make sure to always use lowercase filenames.
+        Preserved: ${existingScript.original} / Discarded: ${filename}`)
+        return
+      }
+    }
+
+    // use this file
+    scriptsByName[cannonicalName] = { original: filename, name, ext }
+  })
+
+  const scripts = Object.values(scriptsByName)
+    .filter((filename): filename is Filename => !!filename)
```

Comment:
Pourquoi est-ce qu'on a besoin du `filename is Filename` ici ?
(aussi... plutôt dur à lire entre les répétitions du mot "filename" et le format array function de mon pdv :/)

Reply:
ca permet de convertir les valeurs en les filtrants et de ne pas avoir de `Filename | undefined` dans le map en dessous